Assets/Character/AaronTestAssets/RunState.cs

```diff
@@ -23,29 +23,29 @@ public class RunState : PlayerState {
 
     public override void ComponentUpdate()
     {
-        Debug.Log("direction: " + direction + ", speed y: " + rgb.velocity.y);
+        Debug.Log("direction: " + direction + ", speed x: " + rgb.velocity.x);
 
-        if (!an.GetCurrentAnimatorStateInfo(0).IsName("Attack"))
+        if (!an.GetCurrentAnimatorStateInfo(0).IsName("Attack") && !an.GetCurrentAnimatorStateInfo(0).IsName("Jump"))
         {
             an.Play("Walk");   
         }
 
 
+
         rgb.AddForce(Vector2.right * direction * play.getSpeed(), ForceMode2D.Impulse);
-        Debug.Log("SLEEPING: " + rgb.IsSleeping());
         Flip();
         
 
-        //if(!grounded)
-        //{
-        //    play.changeState(airborne);
-        //}
+        if(!grounded)
+        {
+            play.changeState(airborne);
+        }
         handleInput();
     }
 
     public override void handleInput()
     {
-        if (Input.GetButtonDown("Backstep"))
+        if (Input.GetButtonDown("Backstep") && grounded)
         {
             backstep.setStartTime(Time.time);
```

Comment:
I recommend having a enterState() function for each state, representing what needs to be initialized when the state transition happens. Then in play.changeState() we call newState.enterState(). This way, you don't need to call backstep.setStartTime(), you can just define backstep.enterState() { startTime = Time.time; } and objects that need to change the player state don't need to know what to initialize themselves.